### ISSUE TITLE
Multiline pasting fails on exactly two lines

### ISSUE THREAD

**Issue by GuiRitter** [microsoft/vscode#80390]
Issue Type: <b>Bug</b>

Hi everyone.

Consider the following piece of text:

```
copy the word "foo" from here ->
foo<- to here
paste 1st here:

paste 2nd here:

paste 3rd here:

```

Copy the `foo` word from arrow to arrow, in order to include the new line character. Place the caret at the end of the line containing `1st`. Place another caret at the end of the line containing `2nd` (default `Alt+click`). Paste. This is the expected result:

```
copy the word "foo" from here ->
foo<- to here
paste 1st here:
foo

paste 2nd here:
foo

paste 3rd here:

```

This is the actual result:

```
copy the word "foo" from here ->
foo<- to here
paste 1st here:

paste 2nd here:foo

paste 3rd here:

```

Only one line receives the paste, without the new line character. The same doesn't happen if there is only 1 caret or more than 2 carets. It also works normally if there's no new line character in the clipboard.

**Note:** this issue was originally opened in [Insiders 1.29.0](https://github.com/microsoft/vscode/issues/62547). It was closed when 1.36 was released, despite I still being able to reproduce it in Insiders 1.37.

Now that 1.38 has been released, since I still can reproduce the issue in Insiders 1.39, I'm reopening this issue.

Thanks in advance.

VS Code version: Code - Insiders 1.39.0-insider (7edc8f786e065edbc9540f1e19ebd9220b9e0cf9, 2019-09-05T05:42:21.072Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-3470 CPU @ 3.20GHz (4 x 3200)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.69GB (1.71GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

**Comment by GuiRitter** [comment]
> (Experimental duplicate detection)
> Thanks for submitting this issue. Please also check if it is already covered by an existing one, like:
> 
> * [Multiline pasting fails on exactly two lines (#62547)](https://www.github.com/microsoft/vscode/issues/62547)
> * [Markdown (preview) nested ordered list rendering error (#79976)](https://www.github.com/microsoft/vscode/issues/79976)
> * [Line Paste between Files bug (#43012)](https://www.github.com/microsoft/vscode/issues/43012)
> * [Pug/vue syntax highlighting fails sometimes (#76373)](https://www.github.com/microsoft/vscode/issues/76373)

Apart from the obvious first one, the other issues appear unrelated to me.

**Comment by alexandrudima** [comment]
Let's track in #80624

**Comment by GuiRitter** [comment]
Ok. But remember that, in this case, when the bug occurs, the line count matches the cursor count.

**Comment by alexandrudima** [comment]
Starting with tomorrow's build, you can configure `"editor.multiCursorPaste": "full"` to get the behavior you want...

**Comment by GuiRitter** [comment]
> Starting with tomorrow's build, you can configure `"editor.multiCursorPaste": "full"` to get the behavior you want...

Strange, I checked in Insiders 1.39 a few days later but that option wasn't available. Now that 1.39 has been released, I can see the option.

Also, only now I realized this option works e.g. to copy a single block of 3 lines of text and paste each line on 3 separate cursors. Different from that, I was only interested in copying multiple selections.

Speaking of which, multiple selections where the number of selections is different from 2 works the same regardless of the value of `"editor.multiCursorPaste"`.

As for the number of selections equal to 2, it's bugged as described for `spread` and working as intended for `full`. Also, different from what was written, `full` is what I expect to be the default behaviour.

So it's finally fixed for me.
